### PR TITLE
dealii: add optional python bindings

### PR DESCRIPTION
**Pull request by davydden**
@jppelteret ping.

### CODE REVIEW

**Comment by jppelteret** [comment]
Thanks @davydden . You're starting to develop an impressive list of build permutations! LGTM.

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 1]
```diff
@@ -51,14 +51,21 @@ class Dealii(Package):
...
-    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",     when='~mpi')  # NOQA: ignore=E501
-    depends_on("boost@1.59.0:+mpi+thread+system+serialization+iostreams", when='+mpi')  # NOQA: ignore=E501
+    # +python won't affect @:8.4.1
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",            when='@:8.4.1~mpi')         # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi",        when='@:8.4.1+mpi')         # NOQA: ignore=E501
+    # since @8.5.0: (and @develop) python bindings are introduced:
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",            when='@8.5.0:~mpi~python')  # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi",        when='@8.5.0:+mpi~python')  # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+python",     when='@8.5.0:~mpi+python')  # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi+python", when='@8.5.0:+mpi+python')  # NOQA: ignore=E501
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dealii/package.py]: You can remove all of those NOQA's. I added `depends_on` to the list of exceptions for line length checks in https://github.com/LLNL/spack/commit/ea80c8d3c3a0690dcb357840cfb0db824f44fdc9

Comment (davydden) [inline var/spack/repos/builtin/packages/dealii/package.py]: cool, thanks for letting me know!

**Comment by davydden** [comment]
@Rombur FYI, i added a variant to build deal.ii with bindings already.

**Comment by Rombur** [comment]
Thanks!!

**Pull request merged**